### DISCUSSION TITLE
How do I disable the default copy on selection?

### DISCUSSION THREAD

**Discussion by eigan** [wezterm/wezterm#3760] [Q&A] [answered]
I move between terminal and other apps frequently and its very annoying that it removes my copied text. I no longer have control of what I have copied.

I tried to look into https://wezfurlong.org/wezterm/config/mouse.html#configuring-mouse-assignments and read somewhere that "CompleteSelection" is in control of the copy functionality? Even with this information, I have no idea how to disable it.

**Comment by nekowinston** [comment, marked as answer]
This should work:
```lua
config.mouse_bindings = {
  {
    event = { Up = { streak = 1, button = "Left" } },
    mods = "NONE",
    action = wezterm.action.Nop,
  },
}
```

**Reply by eigan** [reply]
Works. Thank you!

**Reply by dantefromhell** [reply]
I wonder, how does `wezterm.action.Nop` compare to `wezterm.action.DisableDefaultAssignment` as described in https://wezfurlong.org/wezterm/config/lua/keyassignment/DisableDefaultAssignment.html

**Reply by dantefromhell** [reply]
Answering my own question, as it's documents for `wezterm.action.Nop` [1] 

> If instead of this you want the key presses to pass through to the terminal, look at DisableDefaultAssignment.

[1] https://wezfurlong.org/wezterm/config/lua/keyassignment/Nop.html

**Reply by slmjkdbtl** [reply]
Is there away to disable the specific behavior of copy on select, instead of disabling left mouse up entirely? Cuz this is also disabling clicking on links which I kinda want

**Reply by nekowinston** [reply]
@slmjkdbtl replace `Nop` with `OpenLinkAtMouseCursor`. The [default](https://wezfurlong.org/wezterm/config/mouse.html#default-mouse-assignments) is `CompleteSelectionOrOpenLinkAtMouseCursor("ClipboardAndPrimarySelection")`, so this just gets rid of the clipboard part of the action.

**Reply by slmjkdbtl** [reply]
Thank you, that does give the desired behavior!

**Reply by et304383** [reply]
[=This does not work on MacOS.  I'm still getting copy to clipboard on select.

Edit: what I meant to say is that this does not stop double click from doing copy.  I do not want copy on double click either.

ChatGPT helped me answer the question:

```
config.mouse_bindings = {
  {
    event = { Up = { streak = 1, button = "Left" } },
    mods = "NONE",
    action = wezterm.action.OpenLinkAtMouseCursor,
  },
  {
    event = { Up = { streak = 2, button = "Left" } },
    mods = "NONE",
    action = wezterm.action.Nop,
  },
  {
    event = { Up = { streak = 3, button = "Left" } },
    mods = "NONE",
    action = wezterm.action.Nop,
  },
}
